src/components/App.js

```diff
@@ -0,0 +1,65 @@
+import "../styles/App.css";
+import React, { useState, useEffect } from "react";
+import LocationDetails from "./LocationDetails";
+import ForecastSummaries from "./ForecastSummaries";
+import ForecastDetails from "./ForecastDetails";
+import getForecast from "../requests/getForecast";
+import SearchForm from "./SearchForm";
+
+const App = () => {
+  const [searchText, setSearchText] = useState("");
+  const [forecasts, setForecasts] = useState([]);
+  const [location, setLocation] = useState({ city: "", country: "" });
+  const [selectedDate, setSelectedDate] = useState(0);
+
+  const selectedForecast = forecasts.find(
+    (forecast) => forecast.date === selectedDate
+  );
+
+  const handleForecastSelect = (date) => {
+    setSelectedDate(date);
+  };
+
+  useEffect(() => {
+    getForecast(searchText, setSelectedDate, setForecasts, setLocation);
+  }, []);
+
+  const handleCitySearch = () => {
+    getForecast(searchText, setSelectedDate, setForecasts, setLocation);
+  };
+
+  return (
+    <div className="weather-app">
+      <LocationDetails city={location.city} country={location.country} />
+      <SearchForm
+        searchText={searchText}
+        setSearchText={setSearchText}
+        handleCitySearch={handleCitySearch}
+      />
+      <ForecastSummaries
+        forecasts={forecasts}
+        onForecastSelect={handleForecastSelect}
+      />
+      {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
+    </div>
+  );
+};
+
+// App.propTypes = {
+//   forecasts: PropTypes.arrayOf(
+//     PropTypes.shape({
+//       date: PropTypes.number,
+//       description: PropTypes.string,
+//       icon: PropTypes.number,
+//       temperature: PropTypes.shape({
+//         max: PropTypes.number,
+//         min: PropTypes.number,
+//       }),
+//     })
+//   ).isRequired,
+//   location: PropTypes.shape({
+//     city: PropTypes.string,
+//     country: PropTypes.string,
+//   }).isRequired,
+// };
```

Comment:
Should this be commented? If so, a comment with an explanation as to why would be helpful. Otherwise commented code should never make it into a PR